egs/wsj/s5/steps/nnet3/xconfig_to_configs.py

```diff
@@ -0,0 +1,241 @@
+          '# ' + ' '.join(sys.argv) + '\n'
+          '#It contains the same content as ./xconfig but it was parsed and\n'
+          '#default config values were set.\n'
+          '# See also ./xconfig.expanded.2\n', file=xconfig_file_out)
+
+    for layer in all_layers:
+        print(str(layer), file=xconfig_file_out)
+    xconfig_file_out.close()
+
+    try:
+        xconfig_file_out = open(config_dir + '/xconfig.expanded.2', 'w')
+    except:
+        sys.exit('{0}: error opening file {1}/xconfig.expanded.2 for output'.format(
+                sys.argv[0], config_dir))
+
+    print('# This file was created by the command:\n'
+          '# ' + ' '.join(sys.argv) + '\n'
+          '# It contains the same content as ./xconfig but it was parsed,\n'
+          '# default config values were set, and Descriptors (input=xxx) were normalized.\n'
+          '# See also ./xconfig.expanded.1\n\n',
+          file=xconfig_file_out)
+
+    for layer in all_layers:
+        layer.NormalizeDescriptors()
+        print(str(layer), file=xconfig_file_out)
+    xconfig_file_out.close()
+
+
+
+
+# This function returns a map from config-file basename
+# e.g. 'init', 'ref', 'layer1' to a documentation string that goes
+# at the top of the file.
+def GetConfigHeaders():
+    ans = defaultdict(str)  # resulting dict will default to the empty string
+                            # for any config files not explicitly listed here.
+    ans['init'] = ('# This file was created by the command:\n'
+                   '# ' + ' '.join(sys.argv) + '\n'
+                   '# It contains the input of the network and is used in\n'
+                   '# accumulating stats for an LDA-like transform of the\n'
+                   '# input features.\n');
+    ans['ref'] = ('# This file was created by the command:\n'
+                  '# ' + ' '.join(sys.argv) + '\n'
+                  '# It contains the entire neural network, but with those\n'
+                  '# components that would normally require fixed vectors/matrices\n'
+                  '# read from disk, replaced with random initialization\n'
+                  '# (this applies to the LDA-like transform and the\n'
+                  '# presoftmax-prior-scale, if applicable).  This file\n'
+                  '# is used only to work out the left-context and right-context\n'
+                  '# of the network.\n');
+    ans['all'] = ('# This file was created by the command:\n'
+                  '# ' + ' '.join(sys.argv) + '\n'
+                  '# It contains the entire neural network.  It might not be used\n'
+                  '# in the current scripts; it\'s provided for forward compatibility\n'
+                  '# to possible future changes.\n')
+
+    # Note: currently we just copy all lines that were going to go to 'all', into
+    # 'layer1', to avoid propagating this nastiness to the code in xconfig_layers.py
+    ans['layer1'] = ('# This file was created by the command:\n'
```

Comment:
Maybe modifying the training scripts is the right thing to do.  We could have them look for 'all.config' if layer1.config is absent.